pandas/tests/io/test_sql.py

```diff
@@ -1373,6 +1373,67 @@ def insert_on_conflict(table, conn, keys, data_iter):
         pandasSQL.drop_table("test_insert_conflict")
 
 
+@pytest.mark.parametrize("conn", all_connectable)
+def test_to_sql_on_public_schema(conn, request):
+    if "sqlite" in conn:
+        request.node.add_marker(
+            pytest.mark.xfail(
+                reason="sqlite database does not support schemas",
+            )
+        )
+
+    conn = request.getfixturevalue(conn)
+
+    from sqlalchemy.engine import Engine
```

Comment:
Are you sure you need to explicitly create the schema / table this way? I am under the impression that the `to_sql` call should take care of that anyway.

The reason I am somewhat against doing it via `sqlalchemy` is that ADBC connections do not use / required sqlalchemy at all. So this starts to intertwine the dependencies of those libraries in ways that is unnecessary. For example, if you look at the ADBC fixtures they do not use sqlalchemy to create / drop tables

Reply:
Didn't know `to_sql` also creates the table but apparently it does. However, I do see an issue now in the failing pipeline: `data = b"\xff\x19\x04#42000Unknown database 'public'"` and this happens for `mysql_pymysql_engine `.  Maybe I can add a statement `CREATE DATABASE some_db;` and execute it with sqlalchemy and clean up at the end? wdyt?

Reply:
You shouldn't need to manually construct the `CREATE DATABASE` statement. It looks like I was wrong before about which databases have schemas versus those that do not, and apparently MySQL does not have the same concept of what a schema is as postgresql. 

https://stackoverflow.com/a/11618350/621736

So I think you can go back to @jorisvandenbossche original suggestion and just use the `postgresql_adbc_conn` fixture instead of the all fixture